### PR TITLE
Broadcom pi zero2w neopixel misbehaving/crash fix

### PR DESCRIPTION
**Pull request by RetiredWizard**
These changes fix adafruit/circuitpython#5947. I've tested on both the zero2w and the pi4b (The 4b didn't exhibit the original issue) and the boards now behave properly with 1 to 30 pixels. I've run a 30 pixel animation on the zero2w for 8+ hours without a hang.

**Original corrupted rgb transmission:**
image
**Early Fix, not using transmission queue:**
image
**This PR:**
image

### CODE REVIEW

**Comment by RetiredWizard** [comment]
I just tested this on a zero w rather than the zero2w and it works but it looks like the code is getting stuck more frequently than the other tested board in the loop that checks for an empty queue. I'm not sure that's a problem if I reduce the loop abort counter but I want to test on the three models I have. I'll update once I've finished testing.

**Comment by RetiredWizard** [comment]
Tested on 4b, zero2w and zero w. Runs without glitches or crashes.

File: ports/broadcom/common-hal/neopixel_write/__init__.c [hunk 1]
```diff
@@ -45,7 +45,10 @@ void common_hal_neopixel_write(const digitalio_digitalinout_obj_t *digitalinout,
     uint8_t *pixels, uint32_t num_bytes) {
     // Wait to make sure we don't append onto the last transmission. This should only be a tick or
     // two.
-    while (port_get_raw_ticks(NULL) < next_start_raw_ticks) {
+    int icnt;
+    while ((port_get_raw_ticks(NULL) < next_start_raw_ticks) &
```
Comment (tannewt) [inline ports/broadcom/common-hal/neopixel_write/__init__.c:49]:
```suggestion
    while ((port_get_raw_ticks(NULL) < next_start_raw_ticks) &&
```

File: ports/broadcom/common-hal/neopixel_write/__init__.c [hunk 2]
```diff
@@ -92,7 +95,9 @@ void common_hal_neopixel_write(const digitalio_digitalinout_obj_t *digitalinout,
 
         // Wait for the clock to start up.
         COMPLETE_MEMORY_READS;
-        while (CM_PWM->CS_b.BUSY == 0) {
+        icnt = 0;
+        while ((CM_PWM->CS_b.BUSY == 0) & (icnt++ < 1000)) {
```
Comment (tannewt) [inline ports/broadcom/common-hal/neopixel_write/__init__.c:99]:
```suggestion
        while ((CM_PWM->CS_b.BUSY == 0) && (icnt++ < 1000)) {
```

**Review by tannewt — CHANGES_REQUESTED** [review]
Please use `&&` in your conditions instead of `&`. `&` is a bitwise test and `&&` is boolean logic.

**Comment by RetiredWizard** [comment]
So I misunderstood what `COMPLETE_MEMORY_READS` was doing and clearly over used it in the update for the zero w. I've pulled all the calls I added for that update out and I'm testing it now. The only one remaining that I added was at the very end of the code just before the PWM function of the pin is turned off. I feel like that one might be needed but I guess I'll do some tests.
